test/Microsoft.ML.Benchmarks/TextPredictionEngineCreation.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using Microsoft.ML;
+using Microsoft.ML.Benchmarks;
+using Microsoft.ML.TestFramework;
+using Microsoft.ML.Trainers;
+
+namespace micro
+{
+    [CoreJob]
+    public class TextPredictionEngineCreationBenchmark
+    {
+        private MLContext _context;
+        private ITransformer _trainedModel;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _context = new MLContext();
+            var data = _context.Data.LoadFromTextFile<SentimentData>(BaseTestClass.GetDataPath("wikipedia-detox-250-line-data.tsv"), hasHeader: true);
+
+            // Pipeline.
+            var pipeline = _context.Transforms.Text.FeaturizeText("Features", "SentimentText")
+                .AppendCacheCheckpoint(_context)
+                .Append(_context.BinaryClassification.Trainers.SdcaNonCalibrated(
+                    new SdcaNonCalibratedBinaryTrainer.Options { NumberOfThreads = 1 }));
+
+            // Train.
+            var model = pipeline.Fit(data);
+            var modelPath = "temp.zip";
+
+            // Save model. 
+            _context.Model.Save(model, data.Schema, modelPath);
+
+            // Load model.
+            _trainedModel = _context.Model.Load(modelPath, out var inputSchema);
+        }
+
+        [Benchmark]
+        public PredictionEngine<SentimentData, SentimentPrediction> CreatePredictionEngine()
```

Comment:
>CreatePredictionEngine [](start = 68, length = 22)

This improved from ~300 milliseconds to ~850 microseconds.

Reply:
I tried the original benchmark from the issue - 

https://github.com/glennc/Extensions.ML/blob/840211d64e1e165a124fd17020339479aac79231/benchmarks/micro/CreationBenchmarks.cs#L25-L29

Using the "SentimentModel.zip" that is uploaded on the issue. And the performance got 10x worse than originally reported:

|                 Method |    Mean |    Error |   StdDev | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|----------------------- |--------:|---------:|---------:|------------:|------------:|------------:|--------------------:|
| CreatePredictionEngine | 10.92 s | 0.2412 s | 0.5962 s | 101000.0000 |  37000.0000 |   1000.0000 |          1001.46 MB |

Running the same benchmark using the latest `master` branch packages, I am still seeing:

|                 Method |    Mean |    Error |   StdDev | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|----------------------- |--------:|---------:|---------:|------------:|------------:|------------:|--------------------:|
| CreatePredictionEngine | 1.526 s | 0.0218 s | 0.0193 s |  14000.0000 |   6000.0000 |           - |           146.34 MB |

Which is exactly what I saw originally.

I did an ETW perf trace, and I am still seeing `ApplyAllTransformsToData` being called, which is serializing and deserializing the model (using compression), and that is taking up a lot of time:

![image](https://user-images.githubusercontent.com/8291187/55493979-8bae7b00-55ff-11e9-9e12-b6306c24b648.png)